index.js

```diff
@@ -149,6 +150,90 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Adds the Screwdriver webhook to the Github repository
+     * @method _addWebhook
+     * @param  {Object}    config            Config object
+     * @param  {String}    config.scmUri     The SCM URI to add the webhook to
+     * @param  {String}    config.token      Service token to authenticate with Github
+     * @param  {String}    config.webhookUrl The URL to use for the webhook notifications
+     * @return {Promise}                     Resolve means operation completed without failure.
+     */
+    _addWebhook(config) {
+        const searchForHook = (scmInfo, url, page) => {
+            const getHooks = this.breaker.runCommand({
+                action: 'getHooks',
+                token: config.token,
+                params: {
+                    owner: scmInfo.owner,
+                    repo: scmInfo.repo,
+                    page,
+                    per_page: WEBHOOK_PAGE_SIZE
+                }
+            });
+            const filterForHook = getHooks.then((hooks) => {
+                let found = null;
+
+                hooks.some((hook) => {
+                    const currentHook = hoek.reach(hook, 'config.url');
+
+                    if (currentHook === url) {
+                        found = hook;
+                    }
+
+                    return found;
+                });
+
+                return found;
+            });
+
+            return Promise.all([
+                getHooks,
+                filterForHook
```

Comment:
Both your example and how it's presented in this PR work: we can modify the code without changing the test and it will still pass. It was mainly done this way because of [code complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity).

By keeping the cyclomatic complexity down and not nesting callbacks at-will, we avoid this kind of situation:
 
![btjzedw](https://cloud.githubusercontent.com/assets/721148/20857274/e52ae23c-b8da-11e6-9a50-282f1e88a38c.jpg)

Kidding aside, it's the same reason we also moved away from [callback hell](http://callbackhell.com/) and towards Promises: to keep the code shallow.